clash-cores/src/Clash/Cores/EbTb.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE CPP #-}
+
+-- |
+--   Copyright   :  (C) 2024, QBayLogic B.V.
+--   License     :  BSD2 (see the file LICENSE)
+--   Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+--
+--   8b/10b encoding and decoding functions
+module Clash.Cores.EbTb where
+
+import qualified Clash.Cores.EbTb.Decoder as Dec
+import qualified Clash.Cores.EbTb.Encoder as Enc
+import Clash.Prelude
+
+-- | Data type that contains a 'BitVector 8' with the corresponding error
+--   condition of the decode function
+data DataWord
+  = Dw (BitVector 8)
+  | Cw (BitVector 8)
+  | DwError (BitVector 8)
+  | RdError (BitVector 8)
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Function to check whether a 'DataWord' results in a data word
+isDw :: DataWord -> Bool
+isDw (Dw _) = True
+isDw _ = False
+
+-- | Function to check whether a 'DataWord' results in a data word or
+--   control word
+isValidDw :: DataWord -> Bool
```

Comment:
I feel this needs a better name.
I read it as "is valid dataword", ie when this returns `True`, I'd expect is to be valid and be a data word.

Reply:
I mostly want to have a distinction between "is something valid (so none of the error types)" and "is something a data word", but I would say naming it `isValid` is probably _too_ generic. Do you have any suggestions?

Reply:
These 10-bit quantities are often referred to as _symbols_, which seems like a good alternative to _word_.

Reply:
I have now renamed `DataWord` to `Symbol8b10b`. Does this sound better?